### DISCUSSION TITLE
Can copier use commit hashes instead of tags for tracking applied versions?

### DISCUSSION THREAD

**Discussion by sgarfinkel** [copier-org/copier#1135] [Ideas]
Question in title. If it _can’t_ do this, I think it would be a helpful feature. Sometimes for internal projects creating tagged releases is a pain. Would be nice if copier would just use the commit hash instead, maybe in an opt-in way. I can see how requiring users to semantically tag their templates is generally better practice, though.

**Comment by pawamoy** [comment]
You can use the `-r`, `--vcs-ref` option (check the docs). You can pass it a commit hash or even HEAD.

**Comment by yajo** [comment]
If the repo doesn't have tags, or at least those aren't PEP440-parseable, Copier will aways use `HEAD`.

OTOH, if you're into tags, you can even use `--prereleases` to test tagged updates on some places only.

FWIW tags are only required when you need migrations.
